packages/voting-contract/src/lib.rs

```diff
@@ -271,6 +280,25 @@ where
     Ok(ProposalListResponse { proposals: props? })
 }
 
+pub fn list_text_proposals<P>(
+    deps: Deps,
+    env: Env,
+    start_after: Option<u64>,
+    limit: usize,
+) -> StdResult<ProposalListResponse<P>>
+where
+    P: Serialize + DeserializeOwned,
+{
+    let start = start_after.map(Bound::exclusive_int);
+    let props: StdResult<Vec<_>> = text_proposals()
+        .range(deps.storage, start, None, Order::Ascending)
+        .take(limit)
```

Comment:
We tend to both provide defaults and max to limit when doing this.
But I guess you handle that in the calling contracts?